File: src/main/java/org/pl/parser/ast/VarDeclaration.java
package org.pl.parser.ast;


/**
 * Node that enables variable declaration.
 */
public class VarDeclaration implements INode {
    public final String identifier;
    public final INode assignmentExpr;

    public VarDeclaration(String identifier, INode assignmentExpr) {
        this.identifier = identifier;
        this.assignmentExpr = assignmentExpr;
    }

    @Override
    public String toString() {
        return "VarDeclaration{" +
                "identifier='" + identifier + '\'' +
                ", assignmentExpr=" + assignmentExpr +
                '}';
    }
}

File: src/main/java/org/pl/parser/ast/VarDeclaration.kt
package org.pl.parser.ast

/**
 * A single variable declaration.
 */
class VarDeclaration(val identifier: String, val assignmentExpr: INode?, val typeExpr: TypeNode?) {
    override fun toString() = "VarDeclaration{type=$typeExpr, identifier=$identifier, assignment=$assignmentExpr}"
}